src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_InternalsVisibleTo.vb

```diff
@@ -639,5 +639,220 @@ using System.Reflection;
+                    </Project>
+                </Workspace>)
+                state.SendInvokeCompletionList()
+                Await state.AssertSelectedCompletionItem("Dotted1.Dotted2.Assembly.Dotted3")
+                state.SendTab()
+                Await state.WaitForAsynchronousOperationsAsync()
+                state.AssertMatchesTextStartingAtLine(1, "[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(""Dotted1.Dotted2.Assembly.Dotted3")
+            End Using
+        End Function
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.Completion)>
+        <WorkItem(29447, "https://github.com/dotnet/roslyn/pull/29447")>
+        Public Async Function CodeCompletionReplacesExisitingAssemblyNameWithDots_AndPublicKey_LineBreakExampleFromMSDN() As Task
+            ' Source https://msdn.microsoft.com/de-de/library/system.runtime.compilerservices.internalsvisibletoattribute(v=vs.110).aspx
+            Using state = TestState.CreateTestStateFromWorkspace(
+                <Workspace>
+                    <Project Language="C#" AssemblyName="Dotted1.Dotted2.Assembly.Dotted3"/>
+                    <Project Language="C#" CommonReferences="true" AssemblyName="TestAssembly">
+                        <Document FilePath="C.cs">
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("$$Friend1, PublicKey=002400000480000094" +
+                                                              "0000000602000000240000525341310004000" +
+                                                              "001000100bf8c25fcd44838d87e245ab35bf7" +
+                                                              "3ba2615707feea295709559b3de903fb95a93" +
+                                                              "3d2729967c3184a97d7b84c7547cd87e435b5" +
+                                                              "6bdf8621bcb62b59c00c88bd83aa62c4fcdd4" +
+                                                              "712da72eec2533dc00f8529c3a0bbb4103282" +
+                                                              "f0d894d5f34e9f0103c473dce9f4b457a5dee" +
+                                                              "fd8f920d8681ed6dfcb0a81e96bd9b176525a" +
+                                                              "26e0b3")]
+                        </Document>
+                    </Project>
+                </Workspace>)
+                state.SendInvokeCompletionList()
+                Await state.AssertSelectedCompletionItem("Dotted1.Dotted2.Assembly.Dotted3")
+                state.SendTab()
+                Await state.WaitForAsynchronousOperationsAsync()
+                state.AssertMatchesTextStartingAtLine(1, "[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(""Dotted1.Dotted2.Assembly.Dotted3"" +")
+                state.AssertMatchesTextStartingAtLine(2, "                                                              ""0000000602000000240000525341310004000"" +")
+            End Using
+        End Function
```

Comment:
do we have a test with multiple forms of unterminated strings?  i think we need a test where it's unterminated, and has a newline, and one where it is unterminated and the end of the file is on the same line.

Reply:
I added some explicit tests anyway.